modules/@shopify/checkout-sheet-kit/android/src/main/java/com/shopify/reactnative/checkoutsheetkit/CustomCheckoutEventProcessor.java

```diff
@@ -25,36 +25,44 @@ of this software and associated documentation files (the "Software"), to deal
 
 import android.content.Context;
 import android.util.Log;
-import androidx.annotation.NonNull;
 import com.shopify.checkoutsheetkit.*;
 import com.facebook.react.modules.core.DeviceEventManagerModule;
 import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.WritableNativeMap;
 import com.shopify.checkoutsheetkit.pixelevents.PixelEvent;
+import com.shopify.checkoutsheetkit.lifecycleevents.CheckoutCompletedEvent;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 import java.io.IOException;
 
 public class CustomCheckoutEventProcessor extends DefaultCheckoutEventProcessor {
-  private final ReactContext reactContext;
+  private final ReactApplicationContext reactContext;
 
-  public CustomCheckoutEventProcessor(Context context, ReactContext reactContext) {
+  public CustomCheckoutEventProcessor(Context context, ReactApplicationContext reactContext) {
     super(context);
 
     this.reactContext = reactContext;
   }
 
   @Override
-  public void onCheckoutCompleted() {
-    sendEvent("completed", null);
+  public void onCheckoutCompleted(@NotNull CheckoutCompletedEvent event) {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
```

Comment:
We could make `ObjectMapper` a private instance variable and reuse it (here and in `onWebPixelEvent`, rather than creating a new one for each message.

`Mapper instances are fully thread-safe provided that ALL configuration of the instance occurs before ANY read or write calls`

We're not doing any configuration of the mapper here, so no concerns there

Reply:
I've moved this up to a private instance variable. Let me know if that looks correct to you